ccvm/ccvm.go

```diff
@@ -84,23 +86,25 @@ func prepareCreate(ctx context.Context, args *types.CreateArgs) (*workload, *wor
 
 	ws.Mounts = in.Mounts
 	ws.Hostname = args.Name
-	if ws.NoProxy != "" {
-		split := strings.Split(ws.NoProxy, ",")
-		for _, npip := range noproxyips {
-			var i int
-			for i = 0; i < len(split); i++ {
-				if split[i] == npip {
-					break
-				}
-			}
-			if i == len(split) {
-				split = append(split, npip)
-			}
+
+	if ws.NoProxy != "" || ws.HTTPProxy != "" || ws.HTTPSProxy != "" {
+		npSet := map[string]struct{}{
```

Comment:
Are we planning to hold something within this struct in the future?

Reply:
Nope.  This is the most efficient way of implementing a set in Go.